### ISSUE TITLE
QDirStat Copyright Infringement: qZedStat

### ISSUE THREAD

**Issue by shundhammer** [shundhammer/qdirstat#56]
https://github.com/ZeltaDred/qZedStat

Mail to douglas.osborn@dmail.dixie.edu on 2017-02-28:

> Douglas,
> 
> In August 2016, you forked QDirStat to your own GitHub account to use it as a base for a new project you call qZedStat.
> 
> After doing some minor changes like disabling things related to QDirStat's cleanup actions, you saw fit to change my copyright notices - you replaced them with YOUR name and YOUR e-mail address.
> 
> That happened less than two hours (!) after doing some minor changes. The most urgent thing to you was to claim credit for all that code.
> 
> What makes you think you own any copyright of my code?
> 
> What major accomplishments do you see in the small changes you made that would warrant anything like that?
> 
> By being most generous and putting QDirStat under the GPL V2, I granted everybody (including you) the right to view, copy and modify that code to tailor it to your specific needs. But that does NOT mean that you own the code - merely that you are granted some rights that closed-source software denies you.
> 
> Frankly, I feel insulted by this behaviour. Your contribution to the overall software is miniscule, yet the first thing you do is to make great claims about it. And even more insulting (and completely illegal!) is to remove my copyright notice from it all.
> 
> Now, since I believe that you are a student at Dixie U and you still have a lot to learn, I will not let the great big hammer come down on you. Not immediately, that is.
> 
> What I expect from you:
> 
> - Immediate confirmation that you received this notice.
> 
> - You bring my copyright back in all the places in the sources where you removed it (including the toplevel README.md). Something like "parts (c) Douglas Osborn" is okay, but it must be clear that most of this is NOT your work.
> 
> - A formal apology, both via e-mail and in the toplevel README.md of your project.
> 
> 
> If you do that, I will call this issue settled, and that will be it. Otherwise, I will have to take legal action. And it would be a good idea not to test my patience too much.
> 
> 
> Kind regards
> -- 
> Stefan Hundhammer
> QDirStat Author  
> 
> Krelingstr. 4   *   90408 Nürnberg   *   Germany

**Comment by shundhammer** [comment]
Forked qZedStat to https://github.com/huha2/qZedStat since GitHub won't let me fork the fork of my own project to my own account.

Offending commits:

- https://github.com/ZeltaDred/qZedStat/commit/dd1823f1a840837278a2b5aeadae2b977bc2a979

- https://github.com/ZeltaDred/qZedStat/commit/2a2d133c7b63d411aaea489ce30834cfc059c9eb
Removed original copyright with his own in the toplevel README.md

- https://github.com/ZeltaDred/qZedStat/commit/718490bffc08d621682e29ce2a7d676c5931487a
Removed original copyright in "Help -> About" dialog with his own and added a commercial for https://scitools.com

**Comment by shundhammer** [comment]
Related link: http://gpl-violations.org/

**Comment by flurbius** [comment]
you could have just fetched his changes to a branch of your original
repository :-)

**Comment by shundhammer** [comment]
> you could have just fetched his changes to a branch of your original
repository :-)

Well, actually no, because that would not really prove anything should it come to that. And if that other repo would be silently removed, there would be nothing left behind - unless with having a separate fork.

**Comment by shundhammer** [comment]
Douglas answered, confirmed receipt of my note, apologized by mail and in the README.md.

He promised to change the copyright notice in the "About" dialog, too. Once that happens, this whole affair is settled.

**Comment by flurbius** [comment]
ah ok - didnt thinkl of that.  It seems he just didnt understand the way OS licences work (not that I do either really - too much legalese but I understand enough to know you cant claim someone elses work).  

I probably wont be contributing much to QDirstat though I will continue to use it I have decided to do some javascript coding - there are some exciting tools becoming available these days that are making development much easier and extending the access to desktops, mobiles and so on.  As well as the actual language itself has matured significantly.  I had always avoided js  in favour of more strongly typed languages with superior tooling (c# and java actually, c++ opportunities never seemed to present themselves, although I did do a few c++ classes that processed spatial data where I needed the speed.

Anyway thanks a lot for your patience and help in using Git and Qt,  I appreciate your time and that of the rest of the OSS community  I am particularly impressed having come from a microsoft development background. I had always known that OSS existed just not how much, how good it is and how many developers are involved. I will be sure to let you know if I think of anyway QDirStat could be improved.

cheers
